index.js

```diff
@@ -30,44 +23,54 @@ function parseUrl (url) {
 
 class Processor {
   constructor (url, streamResolver, dimensionFunction, maxWidth, includeMetadata) {
-    var params = url;
+    this.initialize(url);
+
+    if (!filenameRe.test(this.filename) && this.filename !== 'info.json') {
+      throw new this.errorClass(`Invalid IIIF URL: ${url}`); // eslint-disable-line new-cap
+    }
+
+    this.streamResolver = streamResolver;
+    this.dimensionFunction = dimensionFunction || this.defaultDimensionFunction;
+    this.maxWidth = maxWidth;
+    this.includeMetadata = !!includeMetadata;
+  }
+
+  initialize (url) {
+    let params = url;
     if (typeof url === 'string') {
       params = parseUrl(params);
     }
     Object.assign(this, params);
+
     if (this.quality && this.format) {
       this.filename = [this.quality, this.format].join('.');
     }
-    this.streamResolver = streamResolver;
+
     this.errorClass = IIIFError;
-    if (!filenameRe.test(this.filename) && this.filename !== 'info.json') {
-      throw new this.errorClass(`Invalid IIIF URL: ${url}`); // eslint-disable-line new-cap
-    }
-    this.dimensionFunction = dimensionFunction || this.defaultDimensionFunction;
-    this.maxWidth = maxWidth;
-    this.includeMetadata = includeMetadata ? true : false;
   }
 
-  async withStream(id, callback) {
-    if (this.streamResolver.length == 2) {
+  async withStream (id, callback) {
+    if (this.streamResolver.length === 2) {
       return await this.streamResolver(id, callback);
     } else {
-      let stream = await this.streamResolver(id);
+      const stream = await this.streamResolver(id);
       return await callback(stream);
     }
   }
 
   async defaultDimensionFunction (id) {
-    return await this.withStream(id, async (stream)=>{ return await probe(stream) });
+    return await this.withStream(id, async (stream) => {
+      return await probe(stream);
+    });
   }
 
   async dimensions () {
-    let fallback = this.dimensionFunction != this.defaultDimensionFunction;
+    const fallback = this.dimensionFunction !== this.defaultDimensionFunction;
 
     if (this.sizeInfo == null) {
       let dims = await this.dimensionFunction(this.id);
       if (fallback && (dims == null)) {
-        console.warn(`Unable to get dimensions for ${this.id} using custom function. Falling back to probe().`)
+        console.warn(`Unable to get dimensions for ${this.id} using custom function. Falling back to probe().`);
         dims = await this.defaultDimensionFunction(this.id);
```

Comment:
Should either of these checks (line 70, 72) also be changed to a strict equal?

Reply:
Yep. I'm not sure why the linter didn't catch those. Human linter FTW.